dashboard/modules/serve/serve_agent.py

```diff
@@ -251,15 +252,16 @@ async def put_all_applications(self, req: Request) -> Response:
                 text=repr(e),
             )
 
-        client = serve_start(
-            detached=True,
-            http_options={
-                "host": config.http_options.host,
-                "port": config.http_options.port,
-                "root_path": config.http_options.root_path,
-                "location": config.proxy_location,
-            },
-        )
+        async with self._controller_lock:
```

Comment:
nit: it's only used for starting
```suggestion
        async with self._controller_start_lock:
```